spec.html

```diff
@@ -7875,6 +7880,251 @@ <h1>Forward Progress</h1>
+
+      <p>
+        This specification does not make any guarantees that any object will be garbage collected.
+        Objects which are unreachable from ECMAScript may be released after long periods
+        of time, or never at all. For this reason, this specification uses the term
+        "may" when describing behaviour triggered by garbage collection.
+      </p>
+
+      <p>
+        The semantics of WeakRef and FinalizationRegistry objects is based on two
+        operations which happen at particular points in time:
+      </p>
+
+      <ul>
+        <li>
+          When WeakRef.prototype.deref is called, the referent (if *undefined* is not returned)
+          is kept alive so that subsequent, synchronous accesses also
+          return the object. This list is reset when synchronous work is done using
+          the ClearKeptObjects abstract operation.
+        </li>
+
+        <li>
+          When an object which is registered with a FinalizationRegistry becomes
+          unreachable, a call of the FinalizationRegistry's cleanup callback may
+          eventually be made, after synchronous ECMAScript execution completes.
+          The FinalizationRegistry cleanup is performed with the
+          CleanupFFinalizationRegistry abstract operation.
+        </li>
+      </ul>
+
+      <p>
+        Neither of these actions (ClearKeptObjects or CleanupFinalizationRegistry)
+        may interrupt synchronous ECMAScript execution. Because embedding
+        environments may assemble longer, synchronous ECMAScript execution runs,
+        this specification defers the scheduling of ClearKeptObjects and
+        CleanupFinalizationRegistry to the embedding environment.
+      </p>
+
+      <p>
+        Some ECMAScript implementations include garbage collector implementations
+        which run in the background, including when ECMAScript is idle. Letting the
+        embedding environment schedule CleanupFinalizationRegistry allows it to resume
+        ECMAScript execution in order to run finalizer work, which may free up held values,
+        reducing overall memory usage.
+      </p>
+    </emu-clause>
+
+    <emu-clause id="sec-liveness">
+      <h1>Liveness</h1>
+
+      <p>
+        For some set of objects _S_, a <dfn>hypothetical WeakRef-oblivious</dfn>
+        execution with respect to _S_ is an execution whereby
+        WeakRef.prototype.deref being called on a WeakRef whose referent is an
+        element of _S_ always returns *undefined*.
+      </p>
+      <emu-note>
+        WeakRef-obliviousness, together with liveness, capture two notions. One,
+        that a WeakRef itself does not keep an object alive. Two, that cycles in
+        liveness does not imply that an object is live. To be concrete, if
+        determining _obj_'s liveness depends on determining the liveness of
+        another WeakRef referent, _obj2_, _obj2_'s liveness cannot assume
+        _obj_'s liveness, which would beg the question.
```

Comment:
I was heartened by this wording, personally. I don't know if everyone will understand it, but it's also in a note and the phrase is not core to understanding the note. I think authors should maintain a bit of poetic license with non-normative content.

Reply:
Works for me.

Reply:
Continuing from @ljharb's comment above since this was where the actually line was.

To get around the confusion, "which would beg the question" -> "which would be circular reasoning" should be fine. Slightly less precise, but gets the point across.